provider/consul_catalog.go

```diff
@@ -92,9 +93,11 @@ func (provider *ConsulCatalog) watchServices(stopCh <-chan struct{}) <-chan map[
 			default:
 			}
 
-			data, meta, err := catalog.Services(opts)
+			// Listening to changes that leads to `passing` state or degrades from it.
+			// The call is used just as a trigger for further actions (intentionally there is no interest in the received data).
+			_, meta, err := health.State("passing", opts)
```

Comment:
@vholovko I still have a doubt on this implementation in case we register services in Consul **without** healthcheck. Will these services trigger this listener ?